### PR TITLE
BBS-53 Continuous Integration setup

### PR DESCRIPTION
**Pull request by danquah**
BBS-53

### CODE REVIEW

File: modules/ding_test/ding_test.module [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 
 /**
  * Implements hook_xautoload().
  *
  * Have xautoload load classes from our vendor-dir.
- *
- * @param \Drupal\xautoload\Adapter\LocalDirectoryAdapter $adapter
- *   An adapter object that can register stuff into the class loader.
  */
-function ding_test_xautoload($adapter) {
```
Comment (kasperg) [inline modules/ding_test/ding_test.module:11]: According to [Drupal Code standards](https://www.uptrends.com/) we should not user fully qualified class names in the code but add a `use` statement at the top of the file instead.

File: drupal.make [hunk 2]
```diff
@@ -19,3 +19,6 @@ projects[ding2][type] = "profile"
 projects[ding2][download][type] = "git"
 projects[ding2][download][url] = "git@github.com:ding2/ding2.git"
 projects[ding2][download][tag] = "7.x-3.0.2"
+; Temporary patch to make ding installable on mysql 5.7, should be removed when
+; upstreamed.
+projects[ding2][patch][] = ding-core-patches/bpi-nullable-primary-key.patch
```
Comment (kasperg) [inline drupal.make:24]: I am not very keen on patching our own code. Why not just make the change?

File: ding2.make [hunk 3]
```diff
@@ -46,6 +46,7 @@ projects[date][version] = "2.8"
 projects[dibs][subdir] = "contrib"
 projects[dibs][version] = "1.0"
 projects[dibs][patch][] = "http://drupal.org/files/dibs-2107389-2.patch"
+projects[dibs][patch][2812891] = "https://www.drupal.org/files/issues/mysql_5.7_compatibility-2812891-2.patch"
```
Comment (kasperg) [inline ding2.make:49]: In general patches should include a comment about why the patch is needed.

File: ding2.make [hunk 4]
```diff
@@ -219,6 +220,7 @@ projects[menu_position][version] = "1.1"
 
 projects[message][subdir] = "contrib"
 projects[message][version] = "1.10"
+projects[message][patch][2051751] = "https://www.drupal.org/files/message-primary_nullable-2051751-7.patch"
```
Comment (kasperg) [inline ding2.make:223]: As above.

File: .scrutinizer.yml [hunk 5]
```diff
@@ -39,4 +39,11 @@ filter:
...
         # Minified JavaScript files
-        - themes/ddbasic/scripts/*.min.js
+        - themes/ddbasic/scripts/*min.js
+        # Third party JavaScript libraries used in modules
+        - modules/ding_webtrends/js/webtrends*.js
+        - modules/ting_search_carousel/js/jquery.*
+        - modules/contrib/jquery_update/replace/ui/external/jquery.cookie*.js
+        - modules/ding_facetbrowser/js/jquery.cookie*.js
+        - modules/ding_toggle_format/js/jquery.cookie*.js
+        - modules/ding_tabroll/js/jquery-ui-tabs-rotate*.js
```
Comment (kasperg) [inline .scrutinizer.yml:49]: :+1:

This should be upstreamed and libraries preferably downloaded via Drush Make and loaded via Libraries API. That is perhaps outside the scope of this PR.

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+	# code First we build Drupal core only. Instead of using the
+	# profile specified in the make file we use the one we have
+	# just build. This way we do not have to update drupal.make
+	# for each build.
+	drush make drupal.make --projects=drupal -y $(DRUPAL_SITE_PATH)
+
+	# As we're building core seperate from ding we're missing a patch in
+	# drupal.make to ding2 itself, apply it manually.
+	# Notice: this patch has to be upstreamed
+	patch -p1 < ding-core-patches/bpi-nullable-primary-key.patch
```
Comment (kasperg) [inline Makefile:28]: This can be removed if we apply the changes directly.

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+# Run ding2 unittests
+circle-run-unit-tests:
+	cd $(DRUPAL_SITE_PATH)/profiles/ding2/modules/ding_test && composer install
+	cd $(DRUPAL_SITE_PATH) && drush en ding_test -y
+	# Run all ding unit-tests
+	mkdir -p $(CIRCLE_TEST_REPORTS)/phpunit
+	# Circleci has a proxy for the php-executable which confuses run-tests.sh,
+	# so we have to specify the full path to the executable.
+	# Tests to be run are referenced as a comma-seperated list of group-names.
```
Comment (kasperg) [inline Makefile:44]: :+1:

File: circle.yml [hunk 8]
```diff
@@ -1,7 +1,9 @@
 machine:
   php:
     # We support PHP 5.4 so choose the latest version supported by Circle.
-    version: 5.4.37
+    #version: 5.4.37
+    # BBS override
+    version:  5.6.17
```
Comment (kasperg) [inline circle.yml:6]: I would prefer just to make the change instead of having an outcommented configuration value.

**Review by kasperg — CHANGES_REQUESTED** [review]
Reviewed.

These changes partly duplicate [what has already been merged to master](https://github.com/ding2/ding2/pull/627/files).

It may be useful for us to cherry pick that commit and then added what is missing from this PR on top.

File: drupal.make (hunk 2 above)
Comment (danquah) [reply drupal.make:24]: I'm good with that, did it mostly in case this was done upstream before we get to merging our changes.

**Comment by danquah** [comment]
As implementing some of the suggested changes required a rebase, i've opened up a new PR with the suggested changes implementer over at #14

**Comment by danquah** [comment]
Closing as we merged an updated version of this branch in #14

**Pull request closed without merging**